### DISCUSSION TITLE
rewrite const to function

### DISCUSSION THREAD

**Discussion by PZhu888** [neuefische/questions#692] [Web: Self-directed learning] [answered]
Hi,  I would like to rewrite the below const to a function, is there anyone who can check it for me, if it is correct? many thanks!
origin:

``` js
const filteredPeople =
    activeCraft === "All"
      ? peopleInSpace
      : peopleInSpace.filter((person) => person.craft === activeCraft);

```

rewrite:

``` js
function  filteredPeople () { 
if (activeCraft === "All") { 
return peopleInSpace;
} peopleInSpace.filter((person) => person.craft === activeCraft);
}

```

**Comment by onemanwenttomow** [comment, marked as answer]
Hi @PZhu888 

Firstly, i updated your question to include code formatting in the markdown to make it easier to read. You can do that with the ` ``` ` before and after the code you want to share. you can then also add a language you would like to format for e.g

` ``` js ` 
// the code
` ``` `

or 

` ``` css `
// css code you want to share
` ``` `

In terms of your question. At the moment your function will only return something if `activeCraft` is "All". Otherwise the function will return `undefined`. 

Here is one possible way you could do it. 

``` js
function filterPeopleByCraft(people, activeCraft) {
  if (activeCraft === "All") {
    return people;
  } else {
    return people.filter((person) => person.craft === activeCraft);
  }
}

const filteredPeople = filterPeopleByCraft(peopleInSpace, activeCraft);
```

By giving the function parameters, we can make the function more flexible. the other major change is that we use the `else` after the if to also return something. 

I hope this helps.

**Comment by PZhu888** [comment]
Thank you very much! Have a nice weekend.
